modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,109 @@
 	var/greyscale_config_worn_taur_paw
 	var/greyscale_config_worn_taur_hoof
 
+	/// has_taur_sprite[icon_state]: Does this icon_state have ANY taur sprite?
+	var/static/list/has_taur_sprite = list()
+
+	/// has_taur_sprite_suit[icon_state]: has_taur_sprite but for /obj/item/clothng/suit
+	var/static/list/has_taur_sprite_suit = list()
+
+/obj/item/clothing/Initialize(mapload)
+	. = ..()
+	handle_taur_sprites()
+
+/* NOTE FOR handle_taur_sprites(): This can be converted to handle things like digi sprites, vox, etc. I just didn't do it because I don't want to refactor the entire
+system for this one pr. ~niko */
+
+/**
+ * Called in /obj/item/clothing/Initialize().
+ *
+ * Quits instantly if mutant_variants has all taur styles (STYLE_TAUR), then checks to see if the clothing's icon state has an alt sprite for taurs. If no, it then checks to see if
+ * the icon state EXPLICITELY has no taur sprites. If no, it then checks 3 DMI files for the icon_state, and if it appears in any of them, has_taur_sprite is set to
+ * true for that icon state, then it sets the more specific list to true for that state, and finally the specific mutant_variants bitflag for the clothing.
+ *
+ * Finally, if all of the conditions are false, has_taur_sprite for that icon state will be set to false.
+ *
+ * Has seperate logic for suits, as they use a different set of DMIs as of now.
+ */
+/obj/item/clothing/proc/handle_taur_sprites()
+
+	if (!(istype(src, /obj/item/clothing/under))) //remove this if we ever make a taur sprite for anything else, this is for performance
+		return
+
+	if (mutant_variants & STYLE_TAUR)
+		return
+
+	if (has_taur_sprite[icon_state])
+		if (GLOB.naga_taur_uniform_sprites[icon_state])
+			mutant_variants |= STYLE_TAUR_SNAKE
+		if (GLOB.horse_taur_uniform_sprites[icon_state])
+			mutant_variants |= STYLE_TAUR_HOOF
+		if (GLOB.pawed_taur_uniform_sprites[icon_state])
+			mutant_variants |= STYLE_TAUR_PAW
```

Comment:
Oh I see you're trying to make this faster.
This isn't expensive enough to be a problem, assoc list lookups are quite fast. way more then you should need here.
Especially since has_taur_sprite will be much larger then any of these other lists ever could be

If you really want to avoid all 4 lookups, you can just do the same thing you did to the other static lists to these, prefill them with appropriate contents once. Profile before you do that though, I doubt it'll make a large difference

Reply:
Without the caching system, the code's significantly simpler and has very similar performance (/obj/item/clothing/proc/handle_taur_sprites                                                        0.004        0.004        0.004        0.003         2988) so I think I'll get rid of it